c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -578,6 +654,67 @@ public:
   }
 };
 
+class HttpOverCapnpConnectResponseImpl final: public kj::HttpService::ConnectResponse, public kj::Refcounted {
+public:
+  HttpOverCapnpConnectResponseImpl(
+      kj::Own<kj::PromiseFulfiller<kj::HttpClient::ConnectRequest::Status>> fulfiller)
+      : fulfiller(kj::mv(fulfiller)) {}
+
+  ~HttpOverCapnpConnectResponseImpl() noexcept(false) {
+    if (fulfiller->isWaiting()) {
+      auto ex = KJ_EXCEPTION(FAILED,
+          "service's connect() implementation never called accept() nor reject()");
+      if (fulfiller->isWaiting()) {
```

Comment:
This `if()` looks redundant, since you just checked the same thing two lines earlier?